src/authorizer.js

```diff
@@ -41,6 +42,86 @@ class Authorizer {
     throw new Error('Unauthorized');
   }
 
+  async getApiKey(apiGateway, clientId) {
+    let apiKey;
+    try {
+      const apiKeys = await apiGateway.getApiKeys({ nameQuery: clientId, includeValues: true, limit: 1 }).promise();
+      apiKey = apiKeys.items[0];
+    } catch (e) {
+      this.logFunction({
+        level: 'ERROR',
+        title: 'FailedToApiKeys',
+        details: 'An error occurred while fetching api keys',
+        clientId: clientId,
+        error: e
+      });
+    }
+
+    if (apiKey && apiKey.id) {
+      return apiKey;
+    }
+    this.logFunction({
+      level: 'INFO',
+      title: 'ApiKeyNotFound',
+      details: 'No api key has been found, attempting to create one.',
+      clientId: clientId
+    });
+
+    return apiGateway.createApiKey({
+      description: `Key for client ${clientId}`,
+      enabled: true,
+      generateDistinctId: true,
+      name: clientId,
+      value: clientId
+    }).promise();
+  }
+
+  async ensurePlanKey(apiGateway, apiKey) {
+    let usagePlanKey;
+    try {
+      usagePlanKey = await apiGateway.getUsagePlanKey({
+        keyId: apiKey.id,
+        usagePlanId: this.configuration.usagePlan
+      }).promise();
+    } catch (e) {
+      this.logFunction({
+        level: 'ERROR',
+        title: 'FailedToGetUsagePlans',
+        details: 'Usage Plan Key Id is not present',
+        usagePlan: this.configuration.usagePlan,
+        error: e });
+    }
+
+    if (usagePlanKey && usagePlanKey.id) {
+      return usagePlanKey;
+    }
+    this.logFunction({
+      level: 'INFO',
+      title: 'ApiKeyNotPresentInUsagePlan',
+      details: 'Api key is not present in the usage plan, attempting to add it.',
+      usagePlan: this.configuration.usagePlan,
+      apiKey: apiKey
+    });
+
+    return apiGateway.createUsagePlanKey({
+      keyId: apiKey.id,
+      usagePlanId: this.configuration.usagePlan,
+      keyType: 'API_KEY'
+    }).promise();
+  }
+
+  async resolveApiKey(identity) {
+    const apiGateway = new aws.APIGateway();
+    const key = await this.getApiKey(apiGateway, identity);
+    await this.ensurePlanKey(apiGateway, key);
```

Comment:
Is there a use case where we'd use this method on its own and not in a conjunction with `getApiKey`?
Also, is there a use case in which we'd use `getApiKey` without `ensurePlanKey`?

Reply:
No

Reply:
Let's call `ensure` from `getApiKey` then. The code will be more readable and there will be less knowledge required from the user in order to use these calls properly.